internal/infrastructure/worker/utils.go

```diff
@@ -0,0 +1,79 @@
+package worker
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"image"
+	"image/jpeg"
+	"image/png"
+	"net/http"
+)
+
+var (
+	errDecodeImage = errors.New("can't decode image")
+	errEncodeImage = errors.New("can't encode image")
+)
+
+/*
+The decodeImage function decodes a byte slice to an image.Image
+and returns the decoded image, the detected
+content type of the input byte slice, and an error (if any).
+
+It takes a byte slice as input and first detects
+the content type of the image using the http.DetectContentType function.
+
+If the content type is supported, the function decodes
+the byte slice to an image using either
+the jpeg.Decode or png.Decode function depending on the content type.
+
+If the content type is not supported,
+the function returns an error with a message that the content type is unsupported.
+*/
+func decodeImage(buf []byte) (image.Image, string, error) {
+	var (
+		img image.Image
+		err error
+	)
+
+	contentType := http.DetectContentType(buf)
+	switch contentType {
+	case "image/jpeg", "image/jpg":
+		img, err = jpeg.Decode(bytes.NewReader(buf))
+	case "image/png":
+		img, err = png.Decode(bytes.NewReader(buf))
+	default:
+		err = fmt.Errorf("%w: []byte to image.Image. Unsupported content-type: %s", errDecodeImage, contentType)
```

Comment:
Error wrapping done wrong. The `err` from decoders are lost.

Should be something like next:
```suggestion
		err = fmt.Errorf("%s: []byte to image.Image. Unsupported content-type: %s: %w", errDecodeImage, contentType, err)
```